store.go

```diff
@@ -446,6 +446,13 @@ type Store interface {
 	// GID maps (if any are defined) don't contain corresponding IDs.
 	ContainerParentOwners(id string) ([]int, []int, error)
 
+	// MountTempFromSource mounts a source directory from the host using
+	// graphdriver and returns the mountpoint
```

Comment:
What is `id` used for?

Reply:
Used to select the location for the mount point.  We want to make sure out mountpoints and upper directory are under containers/storage/overlay-containers/UUID/upper directory.

Reply:
That should be reflected in the godoc.

But why, though?  There's no indication in here of why tying a mount to a particular container is required.

More broadly, why are these methods of lower level drivers when the only driver that does anything is overlay?  Are we dumping this into the lower level drivers because it's where knowledge of the overlay filesystem is currently concentrated?

What's the plan for cases where multiple processes want to use the same host directory as a temporary directory during overlapping periods of time?

Reply:
@vrothberg Asked for this to be driver based, because he wanted support for BTRFS.

Reply:
How about?
```
	// MountTemp creates a temporarily mount of a source directory from the
	// host using graphdriver in the container `id` private storage.
	// MountTemp then mounts up the storage on the system and returns
	// the mountpoint
```